Documentation/Testing/ExtensionTesting.rst

```diff
@@ -2,846 +2,106 @@
+See also :ref:`Writing unit tests <testing-writing-unit>`
 
-.. code-block:: shell
+..  todo: Add examples
 
-   ln -s "$(which greadlink)" "$(dirname "$(which greadlink)")/readlink"
+.. _testing-extensions-functional:
 
-The :file:`runTests.sh` file of enetcache comes with some additional features,
-for example:
-
-*   it is possible to execute `composer update` from within a container
-    using `Build/Scripts/runTests.sh -s composerUpdate`
-*   it is possible to execute unit tests with a several different PHP versions
-    (with the `-p` option). This is available for PHP linting, too (`-s lint`).
-*   Similar to :ref:`Core test execution <testing-core-examples>` it is possible
-    to break point tests using xdebug (`-x` option)
-*   typo3gmbh containers can be updated using `runTests.sh -u`
-*   verbose output is available with `-v`
-*   help is available with `runTests.sh -h`
-
-.. index:: Testing; Github Actions
-
-Github Actions
---------------
-
-With basic testing in place we now want automatic execution of tests whenever something is merged to the repository and if
-people create pull requests for our extension, we want to make sure our carefully crafted test setup actually
-works. We'll use the CI service of Github Actions to take care of
-that. It's free for open source projects.
-In order to tell the CI what to do, create a new workflow file in `.github/workflows/ci.yml <https://github.com/lolli42/enetcache/blob/master/.github/workflows/ci.yml>`__
-
-.. code-block:: yaml
-
-   name: CI
-
-   on: [push, pull_request]
-
-   jobs:
-
-     testsuite:
-       name: all tests
-       runs-on: ubuntu-latest
-       strategy:
-         matrix:
-           php: [ '8.1', '8.2' ]
-           minMax: [ 'composerInstallMin', 'composerInstallMax' ]
-       steps:
-         - name: Checkout
-           uses: actions/checkout@v2
-
-         - name: Composer
-           run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -s ${{ matrix.minMax }}
-
-         - name: Composer validate
-           run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -s composerValidate
-
-         - name: Lint PHP
-           run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -s lint
-
-         - name: Unit tests
-           run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -s unit
-
-         - name: Functional tests with mariadb
-           run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -d mariadb -s functional
-
-         - name: Functional tests with postgres
-           run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -d postgres -s functional
+Functional tests
+================
 
-         - name: Functional tests with sqlite
-           run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -d sqlite -s functional
+Functional tests also execute the code to be tested. The code is run in a
+context as you would find it within a TYPO3 context.
```

Comment:
I would use the term TYPO3 context here. Better use something similar to following:

```
Functional test execute the test code within a fully composed TYPO3 instance (non-composer mode) with configured extensions and configuration, having full dependency
and extension logic on board and Database backend available.
```

The following description should be changed in some way than. However, I would prefer to make that (main) difference to `Unit tests` clear in the first sentence.

Reply:
I tried to apply your changes please have another look